src/Node.php

```diff
@@ -210,6 +214,33 @@ public function getChildTokens() {
     }
 
     /**
+     * Gets array of declared child keys (cached).
+     *
+     * This is used as an optimization when iterating over nodes: For direct iteration
+     * PHP will create a properties hashtable on the object, thus doubling memory usage.
+     * We avoid this by iterating over just the keys instead.
+     *
+     * @return string[]
+     */
+    private function getChildKeys() {
```

Comment:
You mentioned you had a getSubNodeNames function for php-parser. Should we consider making this method public rather than private? And if so, it would make sense to rename to `getChildNames` for consistency.